### ISSUE TITLE
Benchmark suite for Bridgetown

### ISSUE THREAD

**Issue by jaredcwhite** [bridgetownrb/bridgetown#154]
I was looking at this Hugo benchmarking tool: https://github.com/mainsail-org/hugo-speed-tests

Two things come to mind:

(a) it would be really helpful to have a benchmarking tool, if for no other reason than to find ways to improve and test performance and also verify there aren't major performance regressions between releases.
(b) Those Jekyll vs Hugo comparisons make Ruby seem like a total dog and that really irks me. I want to capture brand new benchmarks that show Bridgetown generating _thousands_ of pages in mere seconds on modern hardware and Ruby installations. Also smoking Gatsby + MDX would make for some great headlines. 😁

**Comment by jaredcwhite** [comment]
P.S. Getting this marketing out in advance of/alongside the release of Ruby 3 is critical. I want 2021 to be "the Year of Ruby" and this can be a small step forward along those lines…

**Comment by andrewmcodes** [comment]
@jaredcwhite Any general ideas of what you'd like to benchmark? 

After lots of experimentation it's become pretty clear in my mind that JS is actually what's eating most of the time up especially in development but I see no reason to benchmark webpack. 

I really like profiling but don't ever get the chance to really do it so I'd be interested in helping out here.

**Comment by jaredcwhite** [comment]
A good start might be to submit a PR to this project so we can at least make sure Bridgetown is comparable to Jekyll and Eleventy. Then that can be a baseline for future optimizations.

￼https://github.com/seancdavis/ssg-build-performance-tests

**Comment by andrewmcodes** [comment]
@jaredcwhite I assume leaving webpack out of it?? Aka just the build command. 

If you are wanting to include the webpack time then I have some perf updates we should add first. 

Fun fact, depending on the version of webpack, node-sass blocks the main thread 😖

**Comment by jaredcwhite** [comment]
Yeah I consider frontend code a whole different animal…I don't think they're trying to test that side of things (other than the fact that it's simply part of the mix when you're talking about stuff like Gatsby, etc.)
